pyquil/api/_qvm.py

```diff
@@ -107,12 +108,13 @@ def __init__(
         else:
             raise TypeError("random_seed should be None or a non-negative int")
 
+        self.timeout = timeout
         self._client = client_configuration or QCSClient.load()
         self.connect()
 
     def connect(self) -> None:
         try:
-            version = qvm.api.get_version_info(client=self._client)
+            version = qvm.api.get_version_info(client=self._client, options=QVMOptions(timeout_seconds=self.timeout))
```

Comment:
This new `options` parameter shouldn't be required, but it is. Regardless, we needed to use it here to apply the timeout, so I added it. Created a follow-up issue in [qcs-sdk-rust#315](https://github.com/rigetti/qcs-sdk-rust/issues/315)